src/js/api/auth/login.test.js

```diff
@@ -0,0 +1,43 @@
+import { login } from './login';
+
+// Mocking the fetch function
+global.fetch = jest.fn(() =>
+  Promise.resolve({
+    ok: true,
+    json: () => Promise.resolve({ accessToken: 'mocked_access_token' }),
+  })
+);
+
+// Mocking the storage module
+jest.mock('../../storage/index.js', () => ({
+  save: jest.fn(),
+  load: jest.fn((key) => {
+    if (key === 'token') {
+      return 'mocked_token';
+    }
+    return null;
+  }),
+}));
+
+describe('login function', () => {
+  it('stores a token when provided with valid credentials', async () => {
+    // Call the login function with valid credentials
+    await login('test@example.com', 'password123');
+
+    // Expect fetch to be called with the correct URL and options
+    expect(fetch).toHaveBeenCalledWith(expect.stringContaining('/social/auth/login'), {

```

Comment:
This is a unique approach, and it took me a while to figure out what was going on. Nice work.